control of observer/fake-quant operations

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26516 Per-channel baseline
* **#26520 control of observer/fake-quant operations**
* #26622 Default histogram observer
* #26519 Serialization and range reduction support for Fake Quant/Observer
* #26876 Fake quantization enhancements for QAT/PTQ support- fix tests

Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

raghuramank10000 committed:
control of observer/fake-quant operations

Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fake_quantize.py:
```diff
@@ -4,6 +4,8 @@
 from .observer import default_observer
 from functools import partial
 
+
```
remove empty changes?

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "control of observer/fake-quant operations"


Hooks to enable control of observer and fake quant that can be used by model.apply() to control fake quant during QAT

Differential Revision: [D17491155](https://our.internmc.facebook.com/intern/diff/D17491155/)

[ghstack-poisoned]
